### PR TITLE
KAZOO-4394: update deprecated types

### PR DESCRIPTION
**Pull request by jamesaimonetti**

### CODE REVIEW

File: core/whistle_couch-1.0.0/src/couch_util.erl [hunk 1]
```diff
@@ -132,7 +131,7 @@ db_classification(?KZ_ACDC_DB) -> 'aggregate';
 db_classification(?WH_SERVICES_DB) -> 'aggregate';
 db_classification(?KZ_PORT_REQUESTS_DB) -> 'aggregate';
 db_classification(?KZ_WEBHOOKS_DB) -> 'aggregate';
-db_classification(<<?WNM_DB_PREFIX_L, _Prefix:5/binary>>) -> 'numbers';
+db_classification(<<"numbers/", _Prefix:5/binary>>) -> 'numbers';
```
Comment (fenollp) [inline core/whistle_couch-1.0.0/src/couch_util.erl]: Is this related?

File: deps/merl-28e5b3829168199e8475fa91b997e0c03b90d280/src/merl.erl [hunk 2]
```diff
@@ -28,11 +28,11 @@
 
 %% NOTE: this module may not include merl.hrl!
 
--type tree() :: erl_syntax:syntaxTree().
+-type trees:tree() :: erl_syntax:syntaxTree().
```
Comment (fenollp) [inline deps/merl-28e5b3829168199e8475fa91b997e0c03b90d280/src/merl.erl]: `trees.erl` doesn't exist in Kazoo nor OTP.

File: deps/meck-0.7.2/src/meck.erl [hunk 3]
```diff
@@ -71,7 +69,7 @@
 
 %% Records
 -record(state, {mod :: atom(),
-                expects :: dict(),
+                expects :: dict:dict(),
```
Comment (fenollp) [inline deps/meck-0.7.2/src/meck.erl]: Please try bumping deps to an ulterior version that compiles with Erlang ≥17 instead of patching them.

File: deps/meck-0.7.2/src/meck.erl [hunk 4]
```diff
@@ -271,7 +269,7 @@ history(Mod) when is_atom(Mod) -> call(Mod, history).
 %% @see num_calls/3
 %% @see num_calls/4
 -spec history(Mod::atom(), Pid:: pid() | '_') -> history().
-history(Mod, Pid) when is_atom(Mod), is_pid(Pid) orelse Pid == '_' ->
```
Comment (fenollp) [inline deps/meck-0.7.2/src/meck.erl]: `deps/` are not checked for trailing whitespace. Now you're just pickyyy.

File: deps/nksip-0.4.0/src/nksip_router.erl [hunk 5]
```diff
@@ -477,4 +475,3 @@ pos2name(29) -> nksip_router_29;
 pos2name(30) -> nksip_router_30;
 pos2name(31) -> nksip_router_31;
 pos2name(P) -> list_to_atom("nksip_router_"++integer_to_list(P)).
-
```
Comment (fenollp) [inline deps/nksip-0.4.0/src/nksip_router.erl]: stahp nao

File: .travis.yml [hunk 6]
```diff
@@ -10,7 +10,6 @@ notifications:
 otp_release:
   - 18.1
   - 17.5
-  - R15B03
```
Comment (fenollp) [inline .travis.yml]: Also remove `…ERLC_OPTS +nowarn_deprecated_type +nowarn_deprecated_function…` below

**Comment by fenollp** [comment]
Note: `merl` is part of OTP as of 18.0.

Question: what about people that can't upgrade from R15B03? (does Riak work with OTP ≠ R15B03 nowadays?) Isn't this update a bit gratuitous given the current compatibility?

File: core/whistle_couch-1.0.0/src/couch_util.erl (hunk 1 above)
Comment (jamesaimonetti) [inline core/whistle_couch-1.0.0/src/couch_util.erl]: Simplifying what got included here. We were hard-coding the other two variants, seemed silly to have a `include_lib` for this one variation.

File: deps/meck-0.7.2/src/meck.erl (hunk 3 above)
Comment (jamesaimonetti) [inline deps/meck-0.7.2/src/meck.erl]: We can update deps as we go. Seems out of scope and this change is pretty minimal.

File: deps/meck-0.7.2/src/meck.erl (hunk 4 above)
Comment (jamesaimonetti) [inline deps/meck-0.7.2/src/meck.erl]: Emacs does whitespace removal when I save any file. I didn't do this, or the others, manually.

File: core/whistle_couch-1.0.0/src/couch_util.erl (hunk 1 above)
Comment (fenollp) [inline core/whistle_couch-1.0.0/src/couch_util.erl]: Could you make this part of another PR then? Given that the current one will probably be sitting here for a while…

**Comment by fenollp** [comment]
@jamesaimonetti Can you rebase so that I merge this? I need it

File: deps/couchbeam-0.8.0/Makefile [hunk 7]
```diff
@@ -3,9 +3,7 @@ ROOT = ../..
 REBAR = $(ROOT)/utils/rebar/rebar
 DIALYZER = dialyzer
 
-EBINS = $(shell find $(ROOT)/core -maxdepth 2 -name ebin -print) \
-	$(shell find $(ROOT)/deps -maxdepth 2 -name ebin -print)
-PA = $(foreach EBIN,$(EBINS),-pa $(EBIN))
+PA =
```
Comment (fenollp) [inline deps/couchbeam-0.8.0/Makefile]: `src/couchbeam_attachments.erl:11: can't find include lib "couchbeam/include/couchbeam.hrl"`

`-include_lib` uses `-pa` I beleive. You should add them back and maybe remove some `-I`s.

**Comment by jamesaimonetti** [comment]
@fenollp this is working more nicely

**Comment by fenollp** [comment]
I'll rebase this then merge it!
